diff --git a/components/PressCategoryByName.vue b/components/PressCategoryByName.vue
index 0a5d9b160f41ad2f2de98539a381c0b569c2c35a..a74a095fc17077974daac77ee80a3fedeaf8fc8b 100644
--- a/components/PressCategoryByName.vue
+++ b/components/PressCategoryByName.vue
@@ -1,5 +1,6 @@
 <script lang="ts" setup>
 import type { CategoryList } from 'valaxy'
+import { isCategoryList } from 'valaxy'
 import { computed } from 'vue'
 
 const props = defineProps<{
@@ -9,7 +10,7 @@ const props = defineProps<{
 
 const category = computed(() => {
   const c = props.categories.children.get(props.item)
-  return c
+  return isCategoryList(c) ? c : undefined
 })
 </script>
 
diff --git a/components/PressDocFooterLastUpdated.vue b/components/PressDocFooterLastUpdated.vue
index 4f1add89e8de1ade25d0f3b613734e6c07169065..5f7430a16d93c2ff8f3af7459a01167ba89ca8e5 100644
--- a/components/PressDocFooterLastUpdated.vue
+++ b/components/PressDocFooterLastUpdated.vue
@@ -1,10 +1,11 @@
 <script setup lang="ts">
-import { useData } from 'valaxy'
+import { useData, useFrontmatter } from 'valaxy'
 import { computed, onMounted, ref, watchEffect } from 'vue'
 import { useI18n } from 'vue-i18n'
 
 const { t } = useI18n()
 const { page } = useData()
+const frontmatter = useFrontmatter()
 
 const date = computed(() => new Date(page.value?.lastUpdated || 0))
 const isoDatetime = computed(() => date.value?.toISOString())
@@ -21,7 +22,7 @@ onMounted(() => {
 </script>
 
 <template>
-  <p class="press-lastUpdated text-right">
+  <p v-if="frontmatter.lastUpdated !== false && page?.lastUpdated" class="press-lastUpdated text-right">
     {{ t('tooltip.last_updated') }}:
     <time :datetime="isoDatetime">{{ datetime }}</time>
   </p>
diff --git a/components/PressNavScreenMenu.vue b/components/PressNavScreenMenu.vue
index 60f51eecfec24023313c2e5924e0ae9f7158e6a5..5942809c03261a1143932a723bd814f31457ef6a 100644
--- a/components/PressNavScreenMenu.vue
+++ b/components/PressNavScreenMenu.vue
@@ -15,7 +15,7 @@ const { localeConfig } = useLocaleConfig()
       <PressNavScreenMenuGroup
         v-else
         :text="item.text || ''"
-        :items="item.items"
+        :items="item.items || []"
       />
     </template>
   </nav>
diff --git a/components/ValaxyMain.vue b/components/ValaxyMain.vue
index ec112424a450b4fd16e9fd63b1f8ab59e59af462..e772e1ecdf2b34da44632e3b25598c4215ede136 100644
--- a/components/ValaxyMain.vue
+++ b/components/ValaxyMain.vue
@@ -62,7 +62,7 @@ onContentUpdated(() => {
               <slot name="main-content-before" />
 
               <ValaxyMd class="mx-auto w-full max-w-4xl" :frontmatter="frontmatter">
-                <div v-if="hasSidebar && !isHome && $title" flex items-center justify-between gap-2>
+                <div v-if="hasSidebar && !isHome && $title && frontmatter.pageTitle !== false" flex items-center justify-between gap-2>
                   <h1 :id="$title" tabindex="-1" class="!mt-0">
                     {{ $title }}
                     <a class="header-anchor" :href="`#${$title}`" aria-hidden="true" />
